docs/source/spec-workload-executor.rst

```diff
@@ -96,8 +165,10 @@ Pseudocode Implementation
     # targetDriver is the driver to be tested.
     import { MongoClient } from "targetDriver"
 
-    # The workloadRunner function accepts a connection string and a stringified JSON blob describing the driver workload.
-    # This function will be invoked with arguments parsed from the command-line invocation of the workload executor script.
+    # The workloadRunner function accepts a connection string and a
```

Comment:
I think this section needs to be updated to clarify what parts of the unified test runner drivers can reasonably expect to re-use and what they cannot.
For instance, you need to add comments specifying what to do with the `createEntities` field. I think the workload executor needs to take the connection string passed in by astrolabe as a string, add to it the uriOptions specified in the driverWorkload blob (read/writrePreferences etc), configure the listeners mandated by this spec (CMAP and Command) and then create the client entity. All remaining entity creation (for collection, database, session etc) can default to the normal unified format behavior.

Reply:
uriOptions are no longer handled by astrolabe, I am investigating removing the other requirements as well.

Reply:
> configure the listeners mandated by this spec (CMAP and Command) and then create the client entity.

Do these really need to be done _before_ the client is created? I would think that command listeners could just be setup around the loop itself. As for CMAP, do most drivers allow that to be configured when constructing the MongoClient, since you might want a listener to pick up pool events from the very beginning?

Reply:
@prashantmital: Can this be resolved? https://github.com/mongodb/specifications/pull/903 added options on the client entity definition.